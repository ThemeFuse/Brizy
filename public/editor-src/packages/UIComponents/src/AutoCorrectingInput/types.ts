import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
