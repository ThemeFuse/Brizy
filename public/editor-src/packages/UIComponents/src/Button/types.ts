import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
