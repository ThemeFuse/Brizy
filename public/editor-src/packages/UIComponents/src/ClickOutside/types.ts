import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
