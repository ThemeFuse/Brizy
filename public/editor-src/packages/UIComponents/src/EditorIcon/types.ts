export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
