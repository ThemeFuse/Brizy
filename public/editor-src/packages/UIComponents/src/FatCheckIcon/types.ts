import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
