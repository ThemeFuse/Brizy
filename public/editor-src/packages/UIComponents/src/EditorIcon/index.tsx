import Loadable from "@loadable/component";
import classnames from "classnames";
import React, { CSSProperties, FC, MouseEvent, ReactElement } from "react";
import { IconsName } from "./types";

const LoadableIcon = Loadable(async () => {
  const { Icon } = await import("./components/Icon");
  return Icon;
});
export interface Props {
  className?: string;
  icon: IconsName;
  style?: CSSProperties;
  onClick?: (e: MouseEvent<SVGElement>) => void;
}

export const EditorIcon: FC<Props> = ({
  className,
  icon,
  style,
  onClick
}): ReactElement => {
  const _className = classnames(
    "brz-icon-svg brz-ed-icon-svg align-[initial]",
    className
  );

  return (
    <LoadableIcon
      name={icon}
      className={_className}
      style={style}
      onClick={onClick}
    />
  );
};
