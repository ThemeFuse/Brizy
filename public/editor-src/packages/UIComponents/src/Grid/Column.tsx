import classNames from "classnames";
import React, { FC } from "react";
import { ColumnProps } from "./types";

const alignMap = {
  start: "brz-ed-control__grid__column--align-start",
  center: "brz-ed-control__grid__column--align-center",
  end: "brz-ed-control__grid__column--align-end"
};

export const Column: FC<ColumnProps> = ({ className, children, align }) => {
  const _className = classNames(
    "brz-ed-control__grid__column",
    className,
    alignMap[align]
  );

  return <div className={_className}>{children}</div>;
};
