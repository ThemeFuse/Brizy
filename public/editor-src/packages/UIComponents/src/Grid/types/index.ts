import { ReactElement } from "react";

export interface ColumnProps {
  className?: string;
  align: "start" | "center" | "end";
}

export interface GridProps {
  className?: string;
  grid: Array<"auto" | number>;
  showSeparator?: boolean;
  children: ReactElement<ColumnProps>[];
}
