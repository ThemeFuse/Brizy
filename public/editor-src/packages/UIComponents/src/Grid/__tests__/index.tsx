import { render } from "@testing-library/react";
import React from "react";
import { Button } from "../../Button";
import { IconsName } from "../../EditorIcon/types";
import { Column } from "../Column";
import { Grid } from "../index";

describe("Test Grid", () => {
  const onClick = jest.fn();

  describe("Snapshots", () => {
    test("Grid with at least 2 columns with 1fr with className", () => {
      const grid = render(
        <Grid grid={[1, 1]} className="test-classname" showSeparator={false}>
          <Column align="start">Column with text align start</Column>
          <Column align="start">Column with text align start</Column>
        </Grid>
      );

      expect(grid.container).toMatchSnapshot();
    });

    test("Grid with at least 2 columns with 1fr w/out className", () => {
      const grid = render(
        <Grid grid={[1, 1]} showSeparator={false}>
          <Column align="start">Column with text align start</Column>
          <Column align="start">Column with text align start</Column>
        </Grid>
      );

      expect(grid.container).toMatchSnapshot();
    });

    test("Grid with at least 2 columns with 1fr with separator", () => {
      const grid = render(
        <Grid grid={[1, 1]} showSeparator={true}>
          <Column align="start">Column with text align start</Column>
          <Column align="start">Column with text align start</Column>
        </Grid>
      );

      expect(grid.container).toMatchSnapshot();
    });

    test("Grid with at least 2 columns:first one 'auto', second one 1fr w/out separator", () => {
      const grid = render(
        <Grid grid={["auto", 1]} showSeparator={false}>
          <Column align="start">Column with text align start</Column>
          <Column align="start">Column with text align start</Column>
        </Grid>
      );

      expect(grid.container).toMatchSnapshot();
    });

    test("Grid with at least 2 columns with align center with 1fr w/out separator", () => {
      const grid = render(
        <Grid grid={[1, 1]} showSeparator={false}>
          <Column align="center">Column with text align center</Column>
          <Column align="center">Column with text align center</Column>
        </Grid>
      );

      expect(grid.container).toMatchSnapshot();
    });

    test("Grid with at least 2 columns with align end with 1fr w/out separator", () => {
      const grid = render(
        <Grid grid={[1, 1]} showSeparator={false}>
          <Column align="end">Column with text align end</Column>
          <Column align="end">Column with text align end</Column>
        </Grid>
      );

      expect(grid.container).toMatchSnapshot();
    });

    test("Grid with at least 2 columns with align end with 1fr w/out separator with custom button component", () => {
      const grid = render(
        <Grid grid={[1, 1]} showSeparator={false}>
          <Column align="end">Column with text align end</Column>
          <Column align="end">
            <Button
              onClick={onClick}
              align="right"
              icon={IconsName.Duplicate}
              reverse={false}
            >
              Text
            </Button>
          </Column>
        </Grid>
      );

      expect(grid.container).toMatchSnapshot();
    });
  });
});
