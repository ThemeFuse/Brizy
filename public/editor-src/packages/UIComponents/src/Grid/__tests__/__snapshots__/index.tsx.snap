// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Grid Snapshots Grid with at least 2 columns with 1fr w/out className 1`] = `
<div>
  <div
    class="brz-ed-control__grid"
    style="grid-template-columns: 1fr 1fr;"
  >
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
  </div>
</div>
`;

exports[`Test Grid Snapshots Grid with at least 2 columns with 1fr with className 1`] = `
<div>
  <div
    class="brz-ed-control__grid test-classname"
    style="grid-template-columns: 1fr 1fr;"
  >
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
  </div>
</div>
`;

exports[`Test Grid Snapshots Grid with at least 2 columns with 1fr with separator 1`] = `
<div>
  <div
    class="brz-ed-control__grid brz-ed-control__grid--separator"
    style="grid-template-columns: 1fr 1fr;"
  >
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
  </div>
</div>
`;

exports[`Test Grid Snapshots Grid with at least 2 columns with align center with 1fr w/out separator 1`] = `
<div>
  <div
    class="brz-ed-control__grid"
    style="grid-template-columns: 1fr 1fr;"
  >
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-center"
    >
      Column with text align center
    </div>
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-center"
    >
      Column with text align center
    </div>
  </div>
</div>
`;

exports[`Test Grid Snapshots Grid with at least 2 columns with align end with 1fr w/out separator 1`] = `
<div>
  <div
    class="brz-ed-control__grid"
    style="grid-template-columns: 1fr 1fr;"
  >
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-end"
    >
      Column with text align end
    </div>
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-end"
    >
      Column with text align end
    </div>
  </div>
</div>
`;

exports[`Test Grid Snapshots Grid with at least 2 columns with align end with 1fr w/out separator with custom button component 1`] = `
<div>
  <div
    class="brz-ed-control__grid"
    style="grid-template-columns: 1fr 1fr;"
  >
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-end"
    >
      Column with text align end
    </div>
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-end"
    >
      <div
        class="brz-ed-control__button brz-ed-control__button-right"
      >
        <span
          class="brz-ed-control__label"
        >
          Text
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Grid Snapshots Grid with at least 2 columns:first one 'auto', second one 1fr w/out separator 1`] = `
<div>
  <div
    class="brz-ed-control__grid"
    style="grid-template-columns: auto 1fr;"
  >
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
    <div
      class="brz-ed-control__grid__column brz-ed-control__grid__column--align-start"
    >
      Column with text align start
    </div>
  </div>
</div>
`;
